### PR TITLE
Add entitlement tests part2

### PR DESCRIPTION
**Pull request by n4ss**
Follow-up PR with `security.*` and `host.*` entitlements tests.

I tried to backport all comments from the [first PR](https://github.com/docker/libentitlement/pull/45).

New:
- I moved the default Moby mounts (which are the same as libcontainers') to a shared package between test utils, entitlements and entitlements tests.
- I added some helpers for OCI Mount objects checks
- Implemented `security.*` and `host.*` entitlements tests

This should close https://github.com/docker/libentitlement/issues/37

### CODE REVIEW

File: secprofile/oci-profile.go [hunk 1]
```diff
@@ -118,7 +119,7 @@ func allowSyscallWithArgs(seccompProfile *specs.LinuxSeccomp, syscallName types.
 				if syscallName == syscallNameToAllowStr &&
 					((len(syscallArgs) == 0 && len(syscallRule.Args) == 0) ||
 						reflect.DeepEqual(syscallRule.Args, syscallArgs)) {
-					logrus.Errorf("Syscall already added: %s", syscallNameToAllowStr)
+					logrus.Debugf("Syscall already added to Seccomp profile: %s", syscallNameToAllowStr)
```
Comment (ashfall) [inline secprofile/oci-profile.go]: Is it possible to add a test for this case when the syscall is already added to the profile?

File: testutils/ocispec.go [hunk 2]
```diff
@@ -153,6 +153,33 @@ func AreSyscallsBlockedBySeccomp(seccompProfile specs.LinuxSeccomp, syscallNames
...
 
+// isSyscallWithArgsAllowedBySeccomp checks that the provided syscall and args are whitelisted by the seccomp profile
+func isSyscallWithArgsAllowedBySeccomp(seccompProfile specs.LinuxSeccomp, syscallName types.Syscall, syscallArgs []specs.LinuxSeccompArg) bool {
+	return !isSyscallWithArgsBlockedBySeccomp(seccompProfile, syscallName, syscallArgs)
+}
+
+// AreSeccompSyscallsWithArgsAllowed checks that the provided list of syscalls and args are whitelisted by the seccomp profile
+func AreSeccompSyscallsWithArgsAllowed(seccompProfile specs.LinuxSeccomp, syscallsWithArgs map[types.Syscall][]specs.LinuxSeccompArg) bool {
+	for syscallName, syscallArgs := range syscallsWithArgs {
+		if !isSyscallWithArgsAllowedBySeccomp(seccompProfile, syscallName, syscallArgs) {
```
Comment (ashfall) [inline testutils/ocispec.go]: Can we test this?

File: testutils/ocispec.go [hunk 3]
```diff
@@ -153,6 +153,33 @@ func AreSyscallsBlockedBySeccomp(seccompProfile specs.LinuxSeccomp, syscallNames
...
+	}
+
+	return true
+}
+
+// AreSyscallsAllowedBySeccomp checks that the provided syscalls are whitelisted by the seccomp profile
+func AreSyscallsAllowedBySeccomp(seccompProfile specs.LinuxSeccomp, syscallNames []types.Syscall) bool {
+	for _, syscallName := range syscallNames {
+		if !isSyscallWithArgsAllowedBySeccomp(seccompProfile, syscallName, []specs.LinuxSeccompArg{}) {
+			return false
```
Comment (ashfall) [inline testutils/ocispec.go]: Can we test this?

File: testutils/ocispec.go [hunk 4]
```diff
@@ -278,3 +305,27 @@ func AreNamespacesDeactivated(nsList []specs.LinuxNamespace, namespaces []specs.
 
 	return true
 }
+
+// PathListMatchRefMount checks that the path list holds exactly the mount destinations of the provided mount list
+func PathListMatchRefMount(mountPathList []string, refMounts []specs.Mount) bool {
```
Comment (ashfall) [inline testutils/ocispec.go]: Looks like the tests don't cover this -- would it make sense to add a unit test just for this method?

File: secprofile/oci-profile.go (hunk 1 above)
Comment (n4ss) [reply secprofile/oci-profile.go]: will fix.

File: testutils/ocispec.go (hunk 2 above)
Comment (n4ss) [reply testutils/ocispec.go]: you mean `isSyscallWithArgsAllowedBySeccomp` or `AreSeccompSyscallsWithArgsAllowed` ?

File: testutils/ocispec.go (hunk 3 above)
Comment (n4ss) [reply testutils/ocispec.go]: same question, inner or outer func?

File: testutils/ocispec.go (hunk 4 above)
Comment (n4ss) [reply testutils/ocispec.go]: Will do.

Comment (n4ss) [reply testutils/ocispec.go]: (it does)

File: testutils/ocispec.go (hunk 2 above)
Comment (ashfall) [reply testutils/ocispec.go]: The case where the inner conditional `if !isSyscallWithArgsAllowedBySeccomp(seccompProfile, syscallName, syscallArgs)` is true and the return value for `AreSeccompSyscallsWithArgsAllowed` is `false`

File: testutils/ocispec.go (hunk 3 above)
Comment (ashfall) [reply testutils/ocispec.go]: The case where the inner conditional (`if !isSyscallWithArgsAllowedBySeccomp(seccompProfile, syscallName, []specs.LinuxSeccompArg{})`) is true and `AreSyscallsAllowedBySeccomp` returns false.

File: defaults/security-ents.go [hunk 5]
```diff
@@ -102,6 +102,7 @@ func securityViewEntitlementEnforce(profile secprofile.Profile) (secprofile.Prof
 		osdefs.SysPtrace,
 		osdefs.SysPersonality,
 		osdefs.SysMadvise,
+		// FIXME: try with: osdefs.SysPrctl,
```
Comment (riyazdf) [inline defaults/security-ents.go]: did we want to make a tracking issue for this?

File: defaults/host-ents.go [hunk 6]
```diff
@@ -40,49 +40,8 @@ var (
...
-			Type:        "mqueue",
-			Source:      "mqueue",
-			Options:     []string{"nosuid", "noexec", "nodev"},
-		},
-	}
-)
-
 func isAllowedMount(mount specs.Mount) bool {
-	for _, allowedMount := range defaultMobyAllowedMounts {
+	for _, allowedMount := range osdefs.DefaultMobyAllowedMounts {
```
Comment (riyazdf) [inline defaults/host-ents.go]: it'd probably be trivial to add a unit test for this function (and it will likely make the coverage green!), can be a followup

File: testutils/ocispec.go (hunk 4 above)
Comment (riyazdf) [reply testutils/ocispec.go]: can we add more cases?  It seems to only hit the `return true` with empty for loop checking

File: testutils/moby-defaults.go [hunk 7]
```diff
@@ -76,3 +77,27 @@ func getDefaultCapSet() map[types.Capability]bool {
 
 	return capSet
 }
+
+// GetNonDefaultMounts returns a mount set from the provided mount list without default Moby mounts that it may contain
+func GetNonDefaultMounts(mountList []specs.Mount) []specs.Mount {
```
Comment (riyazdf) [inline testutils/moby-defaults.go]: Can we add a test with this with a non-empty mountList?  It seems like it isn't being tested from the coverage report

File: defaults/security-ents.go (hunk 5 above)
Comment (n4ss) [reply defaults/security-ents.go]: Yes. opened here: https://github.com/docker/libentitlement/issues/54

File: testutils/moby-defaults.go (hunk 7 above)
Comment (n4ss) [reply testutils/moby-defaults.go]: done!
